syne_tune/blackbox_repository/blackbox_surrogate.py

```diff
@@ -33,6 +33,13 @@ def transform(self, X):
         return X[self.names]
```

Comment:
Same comment here, can you revert those changes? They are unrelated to import.

Reply:
Hmm, they clean up something and fix a docstring ...

Reply:
You want to change the default 1-NN (which does not work, we know that), you need to change 3 places in the code.

Reply:
To facilitate PRs, it is important to focus on one part whenever the changes are significant (which is the case here). 
For instance, in addition to changing imports this PR also modifies how sagemaker session are initialized and how surrogate default are instantiated. This makes it longer to get merged as those are quite independent changes. 

I would prefer to have the discussion on how we could change the default for surrogates on another PR to merge this one faster.

Reply:
You refer to `on_tuner_save` in  `TrialBackend`?

This is just refactoring some code, marked as "hack", in `Tuner`. You yourself criticize my code for doing that, so you should not criticize me for fixing it on the fly. We should avoid this "if isinstance" stuff in general code, and it helped me avoiding a sagemaker depedence.

It takes more time to have these discussions that just to merge things in. Feel free to clean up "my" code as well, I have no problem with that.

Because if I send you a PR with just a clean-up, you are going to complain this does not map to an issue.

It is general good practice to clean up code on the fly, no matter who originally wrote it.

Reply:
Yes, I am referring to this one. Now, I understood that this change is included because of the missing SM import but I did not guess earlier. I asked about this point to the previous PR yesterday and on this PR too 2 hour ago, it would have been helpful for me to know earlier that this change was done because of the SM import as it was not obvious to me.

What I am saying is that having PRs focused on one point makes them easier and faster to merge. This is a better experience for everyone. If there is a bug or a docstring that needs to be fixed it can certainly be included as long as the PR does not become too complex. But if they are points that are included that are not related and that can be argued, it slows the discussion and makes the review more complicated. For instance, the default_surrogate is not fixing anything critical nor a bug fix.

Regarding criticizing your code, this is not my intent. I try to give explanations and references for my comments (such as PEP to justify the practice of putting comments on top). If one comment is not justified or referenced enough, I will appreciate your feedback on it.

Reply:
OK, sorry as well. Next time, I point out the reason for changes.

Reply:
BTW: I also found the reason why we have the surrogate problems. Pretty subtle